### PR TITLE
Fixed undo app name issue

### LINKED ISSUE #7192: App name should not be part of undo-redo history
### Is your proposal related to a problem?

Right now, undo/redo will undo or redo any changes to the app's name

### Describe the solution you'd like

Undo and redo should not affect app name at all

### PR DESCRIPTION
**Pull request by S-Abdul-Rahman**
Fixed #7192

### CODE REVIEW

**Review by johnsoncherian — APPROVED** [review]

File: frontend/src/Editor/Header/EditAppName.jsx [hunk 1]
```diff
@@ -31,6 +31,19 @@ function EditAppName({ appId, appName = '', onNameChanged }) {
       });
   };
 
+  React.useEffect(() => {
+    const handleKeyDown = (e) => {
```
Comment (Ckbhatia) [inline frontend/src/Editor/Header/EditAppName.jsx:35]: You can move this function outside of this component

**Review by kavinvenkatachalam — APPROVED** [review]

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@S-Abdul-Rahman We can call the function inside the useEffect and we need to remove the event listener when the component is unmounted

**Comment by kavinvenkatachalam** [comment]
@arpitnath Will it be fixed in your PR?

**Comment by S-Abdul-Rahman** [comment]
> @S-Abdul-Rahman We can call the function inside the useEffect and we need to remove the event listener when the component is unmounted

@kavinvenkatachalam  So, are you suggesting that these changes need to be reverted back to the previous ones? In the previous code, I added this function inside useEffect and I removed the event listener when the component unmounted.

**Review by kavinvenkatachalam — APPROVED** [review]

**Comment by arpitnath** [comment]
https://github.com/ToolJet/ToolJet/pull/7451#issuecomment-1735297159
@kavinvenkatachalam Yes

**Pull request merged**